File: conformance/tests/add-data-output/add-data-output.runar.ts
import { StatefulSmartContract, ByteString } from 'runar-lang';

export class DataOutputTest extends StatefulSmartContract {
    count: bigint;
    constructor(count: bigint) { super(count); this.count = count; }

    public publish(payload: ByteString) {
        this.count = this.count + 1n;
        this.addDataOutput(0n, payload);
    }
}

File: conformance/tests/add-data-output/add-data-output.runar.zig
const runar = @import("runar");

pub const DataOutputTest = struct {
    pub const Contract = runar.StatefulSmartContract;

    count: i64 = 0,

    pub fn init(count: i64) DataOutputTest {
        return .{ .count = count };
    }

    pub fn publish(self: *DataOutputTest, payload: runar.ByteString) void {
        self.count = self.count + 1;
        self.addDataOutput(0, payload);
    }
};
